### PR TITLE
Fix bugs and add unit tests in instance_norm_grad_kernel when d_scale and d_bias are nullptr. Modify the code style of full_kernel.cc. Add new data type for concat, elementwise_add, gather, scale, scatter ops. test=kunlun 

### PR DESCRIPTION
**Pull request by ZibinGuo**
### PR types
Bug fixes

### PR changes
Others

### Describe
Fix bugs and add unit tests in instance_norm_grad_kernel when d_scale and d_bias are nullptr. Modify the code style of full_kernel.cc. Add new data type for concat, elementwise_add, gather, scale, scatter ops. test=kunlun

### CODE REVIEW

File: paddle/phi/kernels/xpu/instance_norm_grad_kernel.cc [hunk 1]
```diff
@@ -24,46 +25,89 @@ void InstanceNormGradKernel(const Context& dev_ctx,
...
                             const DenseTensor& saved_variance,
-                            const DenseTensor& y_grad,
+                            const DenseTensor& d_y,
                             float epsilon,
-                            DenseTensor* x_grad,
-                            DenseTensor* scale_grad,
-                            DenseTensor* bias_grad) {
+                            DenseTensor* d_x,
+                            DenseTensor* d_scale,
+                            DenseTensor* d_bias) {
```
Comment (houj04) [inline paddle/phi/kernels/xpu/instance_norm_grad_kernel.cc:32]: 函数的参数名字，和paddle/phi/kernels/instance_norm_grad_kernel.h保持一样比较好吧。

File: paddle/phi/kernels/xpu/instance_norm_grad_kernel.cc [hunk 2]
```diff
@@ -24,46 +25,89 @@ void InstanceNormGradKernel(const Context& dev_ctx,
...
-      saved_variance.data<float>(),
-      scale_grad->data<float>(),
-      bias_grad->data<float>(),
-      n,
-      c,
-      h,
-      w,
-      epsilon,
-      true);
+  DenseTensor scale_tmp;
```
Comment (QingshuChen) [inline paddle/phi/kernels/xpu/instance_norm_grad_kernel.cc:80]: 为什么用DenseTensor, 而不是用RAII_GURAD去分配空间？

File: paddle/phi/kernels/xpu/instance_norm_grad_kernel.cc (hunk 1 above)
Comment (ZibinGuo) [reply paddle/phi/kernels/xpu/instance_norm_grad_kernel.cc:32]: 我是参照了cpu和gpu的接口命名方式，它们都是d_y，d_x，d_scale，d_bias

File: paddle/phi/kernels/xpu/instance_norm_grad_kernel.cc (hunk 2 above)
Comment (ZibinGuo) [reply paddle/phi/kernels/xpu/instance_norm_grad_kernel.cc:80]: 好的 下个PR改为RAII_GURAD分配空间

**Review by houj04 — APPROVED** [review]

**Pull request merged**